### PR TITLE
kallisto: 0.43.1 -> 0.46.0

### PR DESCRIPTION
**Pull request by tadeokondrak**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/science/biology/kallisto/default.nix [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
   };
 
-  nativeBuildInputs = [ cmake ];
-  
+  nativeBuildInputs = [ autoconf cmake ];
+
   buildInputs = [ hdf5 zlib ];
 
   meta = with stdenv.lib; {
     description = "kallisto is a program for quantifying abundances of transcripts from RNA-Seq data";
```
Comment (mmahut) [inline pkgs/applications/science/biology/kallisto/default.nix]: Do you mind making the first letter capital, please?

**Comment by mmahut** [comment]
The build fails for me locally.

```
builder for '/nix/store/67w9r2sfyy0pr8gvys0nxwmlp3bfs93d-kallisto-0.46.0.drv' failed with exit code 2; last 10 log lines:
  Scanning dependencies of target kallisto
  make[2]: *** No rule to make target '../src/../ext/htslib/libhts.a', needed by 'src/kallisto'.  Stop.
  make[2]: *** Waiting for unfinished jobs....
  [ 89%] Building CXX object src/CMakeFiles/kallisto.dir/main.cpp.o
  make[1]: *** [CMakeFiles/Makefile2:130: src/CMakeFiles/kallisto.dir/all] Error 2
  make[1]: *** Waiting for unfinished jobs....
  [ 93%] No install step for 'htslib'
  [ 96%] Completed 'htslib'
  [ 96%] Built target htslib
  make: *** [Makefile:130: all] Error 2
```

@GrahamcOfBorg build

**Comment by tadeokondrak** [comment]
It seems to fail building in parallel sometimes, but builds parallel by default without Nix giving it the options.

File: pkgs/applications/science/biology/kallisto/default.nix [hunk 2]
```diff
@@ -1,25 +1,27 @@
...
+    sha256 = "09vgdqwpigl4x3sdw5vjfyknsllkli339mh8xapbf7ldm0jldfn9";
   };
 
-  nativeBuildInputs = [ cmake ];
-  
+  nativeBuildInputs = [ autoconf cmake ];
+
   buildInputs = [ hdf5 zlib ];
 
+  enableParallelBuild = false;
```
Comment (worldofpeace) [inline pkgs/applications/science/biology/kallisto/default.nix]:
```suggestion
  # Parallel build fails in some cases
  enableParallelBuild = false;
```

**Comment by mmahut** [comment]
@tadeokondrak could you please include the comment as per suggestion above?

**Comment by worldofpeace** [comment]
Thanks @tadeokondrak.

**Pull request merged**